### PR TITLE
Add multi func support

### PR DESCRIPTION
**Pull request by tj**
CC @mthenw.

Still no concept of some project-level stuff like overriding config, or dependencies, but this should be the bulk of it and a good enough start for now. I'll add tests at some point hahha.. but I have to run for the night

### CODE REVIEW

File: _examples/go/project.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (mthenw) [inline _examples/go/project.json]: apex.json? :)

File: cmd/apex/apex.go [hunk 2]
```diff
@@ -23,48 +25,51 @@ var version = "0.1.0"
...
-    -e, --env name=val  Environment variable
-    -a, --async         Async invocation
-    -C, --chdir path    Working directory
-    -y, --yes           Automatic yes to prompts
-    -h, --help          Output help information
-    -v, --verbose       Output verbose logs
-    -V, --version       Output version
+    -l, --log-level level   Log severity level [default: info]
+    -a, --async             Async invocation
+    -C, --chdir path        Working directory
```
Comment (mthenw) [inline cmd/apex/apex.go]: not used

File: function/function.go [hunk 3]
```diff
@@ -254,7 +251,7 @@ func (f *Function) Invoke(event, context interface{}, kind InvocationType) (repl
 		FunctionName:   aws.String(f.Name),
 		InvocationType: aws.String(string(kind)),
 		LogType:        aws.String("Tail"),
-		Qualifier:      aws.String("$LATEST"),
+		Qualifier:      aws.String("current"),
```
Comment (mthenw) [inline function/function.go]: `aws.String(CurrentAlias)`

File: function/function.go [hunk 4]
```diff
@@ -285,7 +282,7 @@ func (f *Function) Invoke(event, context interface{}, kind InvocationType) (repl
 
 // Rollback the function to the previous or specified version.
 func (f *Function) Rollback(version ...string) error {
-	f.log("rollbacking")
+	f.Log.Info("rolling back")
```
Comment (mthenw) [inline function/function.go]: hah "rollbacking" doesn't sound good

File: project/project.go [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+// Deploy functions and their configurations.
+func (p *Project) Deploy(names []string) error {
+	p.Log.Debugf("deploying %d functions", len(names))
+
+	sem := make(semaphore.Semaphore, p.Concurrency)
+	errs := make(chan error)
+
+	go func() {
+		for _, name := range names {
+			name := name
```
Comment (mthenw) [inline project/project.go]: ?
// I'm new to golang, don't know this sorcery :)

File: runtime/runtime.go [hunk 6]
```diff
@@ -25,19 +24,17 @@ type Runtime interface {
 
 // CompiledRuntime is a language runtime requiring compilation.
 type CompiledRuntime interface {
-	// Build the given `target`, which should default to a language
-	// specific convention such as "main.go" when zero.
-	Build(target string) error
+	// Build performs a buid, using language-specific conventions
```
Comment (mthenw) [inline runtime/runtime.go]: typo, "build"

**Comment by mthenw** [comment]
Awesome PR :+1:

File: function/function.go (hunk 3 above)
Comment (tj) [inline function/function.go]: updated!

File: project/project.go (hunk 5 above)
Comment (tj) [inline project/project.go]: just re-declares in the loop, same effect as:

```go
go func(name string){}(name)
```

File: runtime/runtime.go (hunk 6 above)
Comment (tj) [inline runtime/runtime.go]: oops haha fixed

**Pull request merged**